Neos.ContentGraph.DoctrineDbalAdapter/Classes/Domain/Projection/GraphProjector.php

```diff
@@ -1018,29 +1064,46 @@ protected function assignHierarchyRelationToNewParent(HierarchyRelation $hierarc
     /**
      * @param callable $operations
      * @throws \Exception
+     * @throws \Throwable
      */
     protected function transactional(callable $operations): void
     {
         $this->getDatabaseConnection()->transactional($operations);
         $this->emitProjectionUpdated();
     }
 
-    protected function updateNodeWithCopyOnWrite($event, callable $operations)
+    /**
+     * @param ContentRepository\Context\Node\CopyableAcrossContentStreamsInterface $event
+     * @param callable $operations
+     * @return mixed
+     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
+     */
+    protected function updateNodeWithCopyOnWrite(ContentRepository\Context\Node\CopyableAcrossContentStreamsInterface $event, callable $operations)
     {
-        // TODO: do this copy on write on every modification op concerning nodes
-
         // TODO: does this always return a SINGLE anchor point??
-        if (method_exists($event, 'getNodeIdentifier')) {
-            // DEPRECATED case: NodeIdentifier
-            $anchorPointForNode = $this->projectionContentGraph->getAnchorPointForNodeAndContentStream($event->getNodeIdentifier(), $event->getContentStreamIdentifier());
-        } else {
-            $anchorPointForNode = $this->projectionContentGraph->getAnchorPointForNodeAndOriginDimensionSpacePointAndContentStream($event->getNodeAggregateIdentifier(), $event->getOriginDimensionSpacePoint(), $event->getContentStreamIdentifier());
-        }
-
-        if ($anchorPointForNode === null) {
-            // TODO Log error
-            throw new \Exception(sprintf('anchor point for node identifier %s and stream %s not found', $event->getNodeIdentifier(), $event->getContentStreamIdentifier()), 1519681260000);
+        switch (get_class($event)) {
+            case NodeReferencesWereSet::class:
+                /** @var NodeReferencesWereSet $event */
+                $anchorPointForNode = $this->projectionContentGraph->getAnchorPointForNodeAndOriginDimensionSpacePointAndContentStream(
+                    $event->getSourceNodeAggregateIdentifier(),
+                    $event->getSourceOriginDimensionSpacePoint(),
+                    $event->getContentStreamIdentifier()
+                );
+                break;
+            default:
+                if (method_exists($event, 'getNodeIdentifier')) {
```

Comment:
There might be some unused "legacy" events relying on this, but none of them will work anyway tbh since the node identifier is already dropped from the DB
+1 for cleanup ticket

Reply:
perfect, removed :)